apps/smart-contracts/core/test/unit/TokenSender.test.ts

```diff
@@ -269,8 +346,58 @@ describe('=> TokenSender', () => {
       expect(outputToken.transfer).not.called
     })
 
+    it('reverts if sent amount > account limit and caller != recipient', async () => {
+      priceOracle.get.returns(OUTPUT_TOKEN_UNIT) // 1 for 1
+      const outputAmount = await calculateExpectedOutput(TEST_LIMIT.add(1))
+      expect(outputAmount).to.be.eq(TEST_LIMIT.add(1))
+      await outputToken.connect(deployer).mint(tokenSender.address, TEST_LIMIT.add(1))
+
+      await expect(tokenSender.connect(deployer).send(user.address, TEST_LIMIT.add(1)))
+        .revertedWithCustomError(tokenSender, 'AccountLimitExceeded')
+        .withArgs(deployer.address, TEST_LIMIT.add(1))
+    })
+
+    it('reverts if sent amount > account limit and caller = recipient', async () => {
+      priceOracle.get.returns(OUTPUT_TOKEN_UNIT) // 1 for 1
+      const outputAmount = await calculateExpectedOutput(TEST_LIMIT.add(1))
+      expect(outputAmount).to.be.eq(TEST_LIMIT.add(1))
+      await outputToken.connect(deployer).mint(tokenSender.address, TEST_LIMIT.add(1))
+
+      await expect(tokenSender.connect(deployer).send(deployer.address, TEST_LIMIT.add(1)))
+        .revertedWithCustomError(tokenSender, 'AccountLimitExceeded')
+        .withArgs(deployer.address, TEST_LIMIT.add(1))
+    })
+
+    it("adds sent amount to caller's total if <= account limit and caller != recipient", async () => {
```

Comment:
```suggestion
    it("adds sent amount to caller's total if = account limit and caller != recipient", async () => {
```
and separate test for `<` case

same for similar test below